src/main/java/gov/cabinetoffice/gap/applybackend/service/SubmissionService.java

```diff
@@ -497,5 +495,36 @@ public SubmissionSectionStatus handleSectionReview(final String userId,
         submissionRepository.save(submission);
         return sectionStatus;
     }
+
+    public SubmissionDefinition transformApplicationDefinitionToSubmissionDefinition(ApplicationDefinition applicationDefinition, int schemeVersion) throws JsonProcessingException {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final String applicationDefinitionJson = objectMapper.writeValueAsString(applicationDefinition);
+        final SubmissionDefinition definition = objectMapper.readValue(applicationDefinitionJson, SubmissionDefinition.class);
+
+        // If scheme version is > 1 then the ESSENTIAL section needs to be split into 2 new sections: ORGANISATION_DETAILS and FUNDING_DETAILS
+        // At this stage it's fine if the Question arrays are empty since we'll be re-building them based on mandatory question responses.
+        if (schemeVersion > 1) {
+            definition.getSections().removeIf(section -> section.getSectionId().equals(ESSENTIAL_SECTION_ID));
```

Comment:
Once this is merged I can look at properly populating these questions with the MQ responses using the code in the TMI2-358 branch

Reply:
I like this being in the service layer good choice.